circuit-types/src/macro_tests.rs

```diff
@@ -0,0 +1,276 @@
+//! Defines tests for macros in the `circuit_macros` crate. We do this so that we may define the
+//! bulk of the traits, data structures, etc outside of the `circuit-macros` crate; as a proc-macro
+//! crate cannot export non proc-macro items
+
+#[allow(clippy::missing_docs_in_private_items)]
```

Comment:
These are mostly moved from the `circuits` crate. I did introduce a new MPC mocking setup here that is used in two of the tests though